pkg/scm/git/git.go

```diff
@@ -29,6 +29,9 @@ type Git interface {
 	Checkout(repo, ref string) error
 	SubmoduleUpdate(repo string, init, recursive bool) error
 	GetInfo(string) *api.SourceInfo
+	CreateEmptyLocalGitDirectory() (string, error)
+	CreateLocalGitDirectory() (string, error)
+	CreateLocalGitDirectoryWithSubmodule() (string, error)
```

Comment:
-1 for adding these to the `Git` interface. These are used only in tests, why are we making this change?

I see in the commit message:

> - move git functions from test package to git package to protect against import loop

@jim-minter what import loop are you referring to?